server/graphql/requisition/src/mutations/errors.rs

```diff
@@ -16,3 +16,29 @@ impl MaxOrdersReachedForPeriod {
         "Maximum orders reached for program, order type and period"
     }
 }
+
+// response requisition errors
+
+pub struct FinalisedRequisition;
+#[Object]
+impl FinalisedRequisition {
+    pub async fn description(&self) -> &str {
+        "Response requisition has already been finalised"
+    }
+}
+
+pub struct TransferRequisition;
+#[Object]
+impl TransferRequisition {
+    pub async fn description(&self) -> &str {
+        "Cannot delete a response requisition transferred from a request requisition"
+    }
+}
+
+pub struct RequisitionWithShipment;
+#[Object]
+impl RequisitionWithShipment {
+    pub async fn description(&self) -> &str {
+        "Cannot delete a response requisition once a shipment has been generated"
+    }
+}
```

Comment:
These errors won't propagate to UI currently. We need error handling for both response and request requisitions